scripts/irods/test/test_iadmin.py

```diff
@@ -1685,6 +1685,43 @@ def test_iadmin_mkgroup_returns_correct_error_on_duplicate_group__issue_7599(sel
         finally:
             self.admin.run_icommand(['iadmin', 'rmgroup', test_group_name])
 
+    def test_iadmin_exits_nonzero_on_error_cases__issue_7734(self):
+        # Nonexistent command
+        out, err, ec = self.admin.run_icommand(['iadmin', 'nonexistentcommand']);
+        self.assertNotEqual(ec, 0)
+
+        # Bad parameters to modrepl
+        out, err, ec = self.admin.run_icommand(['iadmin', 'modrepl', 'foo', 'bar', 'baz']);
+        self.assertNotEqual(ec, 0)
+
+        # Nonexistent data object 
+        out, err, ec = self.admin.run_icommand(['iadmin', 'modrepl', 'logical_path', '/tempZone/object/does/not/exist', 'replica_number', '0', 'DATA_COMMENTS', 'something']);
+        self.assertNotEqual(ec, 0)
+
+        try:
+            test_file_name = 'test_file_name_issue_7734'
+            self.admin.assert_icommand(['itouch', self.admin.home_collection + '/' + test_file_name])
```

Comment:
This is _almost_ equivalent to:
```py
self.admin.assert_icommand(['itouch', test_file_name])
```

The benefit is that by NOT including the path prefix, the data object is created in `self.admin.session_collection`. Prefer that collection where possible because it is automatically cleaned up by the test harness, meaning the `irm` in the `finally` block can be removed.